x/twap/logic.go

```diff
@@ -268,13 +259,15 @@ func computeTwap(startRecord types.TwapRecord, endRecord types.TwapRecord, quote
 }
 
 // twapLog returns the logarithm of the given spot price, base 2.
-// TODO: basic test
 func twapLog(price sdk.Dec) sdk.Dec {
 	return osmomath.BigDecFromSDKDec(price).LogBase2().SDKDec()
 }
 
-// twapPow exponentiates the geometricTwapMathBase to the given exponent.
-// TODO: basic test and benchmark.
+// twapPow exponentiates 2 to the given exponent.
 func twapPow(exponent sdk.Dec) sdk.Dec {
-	return osmomath.PowApprox(geometricTwapMathBase, exponent, geometricTwapPowPrecision)
+	exp2 := osmomath.Exp2(osmomath.BigDecFromSDKDec(exponent.Abs()))
+	if exponent.IsNegative() {
+		return osmomath.OneDec().Quo(exp2).SDKDec()
+	}
```

Comment:
Note: this is possible if sp0 is < 1

Currently, there is a micro-optimization possible to avoid taking ` 1 / result` twice. This happens when:
```
arithmeticMeanOfLogPrices.IsNegative() && quoteAsset == startRecord.Asset1Denom
```

I'm planning to revisit this in the future

Reply:
tracked in: https://github.com/osmosis-labs/osmosis/issues/3541